### ISSUE TITLE
Wave discriminator?

### ISSUE THREAD

**Issue by turian** [lucidrains/audiolm-pytorch#66]
The original soundlm paper also uses a wave-based discriminator:

III.D
"For the wave-based discriminator, we use the same multiresolution convolutional discriminator proposed in [15] and
adopted in [45]. Three structurally identical models are applied
to the input audio at different resolutions: original, 2-times down-sampled, and 4-times down-sampled. Each single-scale
discriminator consists of an initial plain convolution followed
by four grouped convolutions, each of which has a group size
of 4, a down-sampling factor of 4, and a channel multiplier
of 4 up to a maximum of 1024 output channels. They are
followed by two more plain convolution layers to produce the
final output, i.e., the logits."

citing:  K. Kumar, R. Kumar, T. de Boissiere, L. Gestin, W. Z. Teoh, J. Sotelo,
A. de Brebisson, Y. Bengio, and A. Courville, “MelGAN: Generative
adversarial networks for conditional waveform synthesis,” in Advances
in Neural Information Processing Systems, 2019

M. Tagliasacchi, Y. Li, K. Misiunas, and D. Roblek, “SEANet: A multimodal speech enhancement network,” in Interspeech, 2020, pp. 1126–
1130.

**Comment by lucidrains** [comment]
@turian that's already taken care of [here](https://github.com/lucidrains/audiolm-pytorch/blob/main/audiolm_pytorch/soundstream.py#L537) or am i mistakened?
